src/libpriv/rpmostree-core.c

```diff
@@ -1526,51 +1441,45 @@ break_single_hardlink_at (int           dfd,
 
   if (stbuf.st_nlink > 1)
     {
-      g_autofree char *path_tmp = g_strconcat (path, ".XXXXXX", NULL);
-      if (!renametempat (dfd, path, dfd, path_tmp, error))
-        goto out;
+      guint count;
+      gboolean copy_success = FALSE;
+      char *path_tmp = glnx_strjoina (path, ".XXXXXX");
 
-      if (!glnx_file_copy_at (dfd, path_tmp, &stbuf, dfd, path, 0,
-                              cancellable, error))
-        goto out;
-
-      if (unlinkat (dfd, path_tmp, 0) < 0)
+      for (count = 0; count < 100; count++)
         {
-          glnx_set_prefix_error_from_errno (error, "%s", "unlinkat");
-          goto out;
-        }
-    }
+          g_autoptr(GError) tmp_error = NULL;
 
-  ret = TRUE;
-out:
-  return ret;
-}
+          glnx_gen_temp_name (path_tmp);
 
-static gboolean
-copy_dir_contents_nonrecurse_at (int            src_dfd,
-                                 const char    *srcpath,
-                                 int            dest_dfd,
-                                 GCancellable  *cancellable,
-                                 GError       **error)
-{
-  gboolean ret = FALSE;
-  g_auto(GLnxDirFdIterator) dfd_iter = { FALSE, };
-  struct dirent *dent = NULL;
+          if (!glnx_file_copy_at (dfd, path, &stbuf, dfd, path_tmp, 0,
+                                  cancellable, &tmp_error))
+            {
+              if (g_error_matches (tmp_error, G_IO_ERROR, G_IO_ERROR_EXISTS))
+                continue;
+              g_propagate_error (error, g_steal_pointer (&tmp_error));
+              goto out;
```

Comment:
I swear, my brain saw an if on that previous line before! Or maybe that's just what it wanted to see. :dizzy: